tools/autograd/gen_python_functions.py

```diff
@@ -1541,3 +886,115 @@ def op_name(declaration):
+#
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ #
+
+# These helper functions allow us to call the new codegen API from the
+# old codegen script (which operates on Declarations.yaml).
+
+# TODO: remove all these HACKs after migration is completed!
+
+# function schema str -> NativeFunction
+NF_TABLE: Optional[Dict[str, NativeFunction]] = None
+
+def init(native_yaml_path: str) -> None:
+    from tools.codegen.gen import parse_native_yaml
+    global NF_TABLE
+    NF_TABLE = {str(f.func): f for f in parse_native_yaml(native_yaml_path)}
+
+# Multiple decl entries can map to the same native function (because of deprecated decl).
+def decl_to_native_function(decl: Dict[str, Any]) -> NativeFunction:
+    assert NF_TABLE is not None, 'need to initialize codegen.api.python with init()'
+    function_schema_str = decl['schema_string'][6:]
+    assert function_schema_str in NF_TABLE, f'cannot find func: {function_schema_str}'
+    return NF_TABLE[function_schema_str]
+
+# Each decl entry has unique python signature.
+def decl_to_python_signature(decl: Dict[str, Any], *, method: bool) -> PythonSignature:
+    f = decl_to_native_function(decl)
+
+    @with_native_function
+    def go(f: NativeFunction) -> PythonSignature:
+        return signature(f, method=method)
+
+    python_sig = go(f)
+
+    if decl.get('deprecated', False):
+        # TODO: directly load 'deprecated.yaml'.
+        # deprecated.yaml doesn't have complete type information, we need
+        # leverage the source signature (to which it delegates the call).
+        # Deprecated signature might reorder input_args and input_kwargs,
+        # but never changes output_args nor python_binding_args (if any?),
+        # so here we only look into these two types of args.
+        src_args: Dict[str, PythonArgument] = {a.name: PythonArgument(
+            name=a.name,
+            type=a.type,
+            cpp_type_str=a.cpp_type_str,
+            default=None,
+            default_init=None,
+        ) for a in itertools.chain(python_sig.input_args, python_sig.input_kwargs)}
+        args: List[Dict[str, Any]] = decl['arguments']
+        input_arg_names: List[str] = \
+            list(str(a['name']) for a in args if not a['kwarg_only'] and not a['output'])
+        input_kwarg_names: List[str] = \
+            list(str(a['name']) for a in args if a['kwarg_only'] and not a['output'])
+        python_sig = PythonSignatureDeprecated(
+            name=python_sig.name,
+            input_args=tuple(src_args[n] for n in input_arg_names if not method or n != 'self'),
+            input_kwargs=tuple(src_args[n] for n in input_kwarg_names),
+            output_args=python_sig.output_args,
+            tensor_options_args=python_sig.tensor_options_args,
+            method=python_sig.method,
+            deprecated_args_names=tuple(str(a['name']) for a in args),
+            deprecated_args_exprs=tuple(decl.get('call_args')),
+        )
```

Comment:
Not really sure what's going on here

Reply:
> Not really sure what's going on here

You can check the examples I listed in "PythonSignatureDeprecated" data model - hope it helps you understand the meaning of 'deprecated_args_names' and 'deprecated_args_exprs'.